rust-utils/src/launcher_lair_client.rs

```diff
@@ -300,6 +368,65 @@ pub async fn derive_seed_from_device_bundle(
                 )?,
             })
         }
-        _ => Err(napi::Error::from_reason("Unsupported Cipher".to_string())),
+        _ => Err(napi::Error::from_reason(
+            "Unsupported LockedSeedCipher variant".to_string(),
+        )),
     }
 }
+
+pub async fn unlock_device_bundle(
+    device_bundle: &String,
+    passphrase: String,
+) -> napi::Result<UnlockedSeedBundle> {
+    let locked_bundle = base64::decode(device_bundle).map_err(|e| {
+        napi::Error::from_reason(format!("Failed to decode device bundle: {:?}", e))
+    })?;
+    match UnlockedSeedBundle::from_locked(&locked_bundle)
+        .await
+        .map_err(|e| {
+            napi::Error::from_reason(format!("Failed to create UnlockedSeedBundle: {:?}", e))
+        })?
+        .remove(0)
+    {
+        LockedSeedCipher::PwHash(bundle) => {
+            let passphrase_bufread = BufRead::from(passphrase.as_bytes());
+            let seed = bundle.unlock(passphrase_bufread).await.map_err(|e| {
+                napi::Error::from_reason(format!(
+                    "Failed to unlock cipher with the given passphrase: {:?}",
+                    e
+                ))
+            })?;
+
+            Ok(seed)
+        }
+        _ => Err(napi::Error::from_reason(
+            "Unsupported LockedSeedCipher variant".to_string(),
+        )),
+    }
+}
+
+async fn get_or_create_seed(client: &LauncherLairClient, tag: &str) -> Result<SeedInfo> {
```

Comment:
Perhaps this function could be part of the impl block of `LauncherLairClient` since it ends up getting called with `&self` as the first arg

Reply:
Yes I think that makes sense

Reply:
pushed a commit that should address this